src/main/kotlin/com/nikitanazarov/coroutinestacks/ui/CoroutineFramesList.kt

```diff
@@ -0,0 +1,132 @@
+        private val itemBorder = BorderFactory.createMatteBorder(0, 0, 1, 0, JBColor.GRAY)
+        private val leftPaddingBorder: Border = JBUI.Borders.emptyLeft(Constants.leftPaddingForBorder)
+        private val compoundBorder = BorderFactory.createCompoundBorder(itemBorder, leftPaddingBorder)
+    }
+
+    init {
+        val debugProcess = suspendContext.debugProcess
+
+        val data = mutableListOf<String>()
+        data.add(trace.header)
+        data.addAll(trace.stackFrameItems.map { it.toString() })
+        setListData(data.toTypedArray())
+        val lastStackFrame = data.getOrNull(1)
+
+        border = if (lastRunningStackFrame == lastStackFrame) {
+            createRoundedBorder(JBColor.BLUE)
+        } else {
+            createRoundedBorder(JBColor.BLACK)
+        }
+
+        cellRenderer = object : DefaultListCellRenderer() {
+            override fun getListCellRendererComponent(
+                list: JList<*>,
+                value: Any,
+                index: Int,
+                isSelected: Boolean,
+                cellHasFocus: Boolean
+            ): Component {
+                val renderer = super.getListCellRendererComponent(list, value, index, isSelected, cellHasFocus)
+                if (renderer !is JComponent) {
+                    return renderer
+                }
+
+                with(renderer) {
+                    val listSize = list.model.size
+                    if (index == 0) {
+                        toolTipText = trace.coroutinesActiveLabel
+                        font = font.deriveFont(Font.BOLD)
+                    } else if (index < listSize) {
+                        toolTipText = value.toString()
+                    }
+                    border = when {
+                        index < listSize - 1 -> compoundBorder
+                        index == listSize - 1 -> leftPaddingBorder
+                        else -> null
+                    }
+                }
+
+                return renderer
+            }
+        }
+
+        addMouseListener(object : MouseAdapter() {
+            override fun mouseClicked(e: MouseEvent?) {
+                val list = e?.source as? JBList<*> ?: return
+                val index = list.locationToIndex(e.point).takeIf { it > 0 } ?: return
+                val stackFrameItem = trace.stackFrameItems[index - 1]
+
+                stackFrameItem?.let { frameItem ->
+                    val frame = frameItem.createFrame(debugProcess)
+
+                    if (suspendContext.activeExecutionStack != null && frame != null) {
+                        suspendContext.setCurrentStackFrame(suspendContext.activeExecutionStack, frame)
+                    }
+                }
+            }
+        })
+    }
+
+    // Copied from org.jetbrains.kotlin.idea.debugger.coroutine.view.CoroutineSelectedNodeListener#setCurrentStackFrame
+    private fun SuspendContextImpl.setCurrentStackFrame(executionStack: XExecutionStack?, stackFrame: XStackFrame) {
```

Comment:
Let's make executionStack non nullable and check if it is null on the call site.